components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/index.html

```diff
@@ -1985,54 +2523,167 @@ <h4>Return Attributes: </h4>
                 isMandatory: false,
                 elements: [
                     {
-                        key: 'buffer.size',
+                        name: 'buffer.size',
                         defaultValue: '256',
                         class: 'mandatory-key',
                         isMandatory: true
                     },
                     {
-                        key: 'workers',
+                        name: 'workers',
                         defaultValue: '2',
                         class: 'mandatory-key',
                         isMandatory: true
                     },
                     {
-                        key: 'batch.size.max',
+                        name: 'batch.size.max',
                         defaultValue: '5',
                         class: 'mandatory-key',
                         isMandatory: true
                     },
                 ]
             }
             ],
-            table_predefined_annotations: [
+            type_table_predefined_annotations: [
             {
                 name: "primaryKey",
-                values: [{
-                    value: ""
-                }],
-                isChecked: false
+                optional: true
             },
             {
                 name: "index",
-                values: [{
-                    value: ""
-                }],
-                isChecked: false
+                optional: true
             }
             ],
+            type_aggregation_predefined_annotations: [
+                {
+                    name: "purge",
+                    optional: true,
+                    parameters: [
+                        {
+                            name: "enable",
+                            defaultValue: "true",
+                            description: "Data purging can be enabled or disabled using this option.",
+                            optional: true,
+                            type: ["BOOL"]
+                        },
+                        {
+                            name: "interval",
+                            defaultValue: "10 sec",
+                            description: "The interval to purge data can be mentioned using this option.",
+                            optional: true,
+                            type: ["INT", "LONG", "TIME"]
+                        }
+                    ],
+                    annotations: [{
```

Comment:
why there is another level called annotations, cant we add them to root level ?

Reply:
no because that is the structure of annotation it can have elements and sub annotations